app/forms/candidate_interface/pick_course_form.rb

```diff
@@ -65,6 +65,14 @@ def course
       @course ||= provider.courses.find(course_id)
     end
 
+    def already_picked_this_one?
+      application_form.application_choices.includes([:course]).any? { |application_choice| application_choice.course == course }
```

Comment:
Could you do the comparison on `application_choice.course_id` to avoid needing to `include` course?

Reply:
Not a column on the `application_choice` record unfortunately, the id is obtained via the associated `course_option`.